### DISCUSSION TITLE
"production" use

### DISCUSSION THREAD

**Discussion by lllama** [aio-libs/aiosmtpd#298] [General]
Hello,

I'm implementing a mail service for my team to use. We have some unique requirements around access and accounts, which is why I'm using a custom solution. At the moment I have postfix handling inbound SMTP, which then hands off to Dovecot. Access to mails is then handled by a webapp that connects to Dovecot via IMAP.

I'm thinking that I could replace Dovecot with aiosmtpd running an LMTP server and the maildir handler. The mail service is fairly low volume, and removing Dovecot would (or could) simplify things.

Would aiosmtpd be up to the challenge? My plan is to run it as part of the webapp stack (which will be starlette running under uvicorn). This should let me implement some features around processing mails as they arrive and notifying users.

**Comment by lllama** [comment]
Just going to @waynew while I hunt out a gopher client. More because I'm wondering if there's a better place to ask questions.

**Comment by waynew** [comment]
Hey! I meant to reply to this a while ago but life and all of that.

My experience is that aiostmpd works great - that's actually how I run my own personal mailserver. I did a bit of customization because I actually *don't* want to adhere to some of the standards, but I also want to lie about them to people who send emails.

Anyways, yeah - aiosmptd should be up to the challenge. I haven't stress tested it at all, but mine runs on the smallest DO droplet alongside nginx and hasn't had a single problem. I did have to figure out the mail sending thing with signing for DKIM and all of that, but that was basically just a me thing.

Though also somewhat ironically I've been trying to get Dovecot into the mix for Delta Chat :joy:

**Reply by lllama** [reply]
Great!

At the moment I'm running it as an LMTP server behind postfix as I need to make sure that we accept all incoming mail. (We've had some issues with other servers failing SSL handshakes but I'm pretty sure that was a golang issue.)

I'm tempted now to replace that with just aiosmptd as then it's fewer moving parts. I'll have a play. 

Thanks again.

**Comment by waynew** [comment]
The only problem that I had was that I was erroneously sending `NNN-thing\nXXX-other-thing\n250 HELP` back, instead of `\r\n` and Outlook was very strict about that `\r`. Took a bit of debugging to find. Other than that, SSL seems to work just fine - I have starttls as well as an SSL handler - I'm pretty sure I use the certs that I get from LetsEncrypt. Good luck!
